### PR TITLE
Use Go version, add go.mod, and use go-licenser

### LINKED ISSUE #8: Remove self vendoring
Looks like we accidentally vendored some parts of this library within itself. 

https://github.com/elastic/go-windows/tree/master/vendor/github.com/elastic/go-windows

### PR DESCRIPTION
**Pull request by andrewkroh**
Some light project cleanup.

- Update Travis/AppVeyor to Go 1.12.
- Update GVM on AppVeyor to v0.2.0.
- Add a go.mod and go.sum file.
- Run go generate.
- Update license headers based on go-licenser.
- Remove erroneous vendor directory. Fixes #8.
- Enable windows on travis ci
- Always use LF and never use CRLF

### CODE REVIEW

File: .ci/scripts/check_lint.go [hunk 1]
```diff
@@ -36,30 +39,33 @@ var ignoreWarnings = []string{
...
 
 	out, err = filterIgnores(out)
 	if err != nil {
-		fmt.Fprintln(os.Stderr, "error", err)
-		os.Exit(1)
+		log.Fatal(err)
 	}
 
 	if len(out) > 0 {
+		log.Println("There are golint warnings.")
```
Comment (ruflin) [inline .ci/scripts/check_lint.go]: Why not use log.Fatalf here too?

File: .gitattributes [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# Treat all files in the Go repo as binary, with no git magic updating
+# line endings. Windows users contributing to Go will need to use a
+# modern version of git and editors capable of LF line endings.
+
+* -text
```
Comment (ruflin) [inline .gitattributes]: Do we need this in other repos too?

Comment (andrewkroh) [reply .gitattributes]: CRLF was causing problems when checking goimports on the new Travis CI windows workers. So probably we can add it as needed to the other repos.

File: .ci/scripts/check_lint.go (hunk 1 above)
Comment (andrewkroh) [reply .ci/scripts/check_lint.go]: I wanted to write the error message to stderr, then continue to pass through the output of goimports to stdout, then `exit 1`.

**Pull request merged**